docs/knowledgebase/advanced/try-runtime.md

```diff
@@ -0,0 +1,184 @@
+---
+title: try-runtime
+---
+
+The `try-runtime` tool is built to query a snapshot of runtime storage, using an in-memory-externalities to store state. In this way, 
+it enables runtime engineers to write tests for a specified runtime state.
+Being part of [`sc-cli`][sc-cli-rustdocs], it is designed to be used as a command line interface to specify 
+what block to capture and which pallets to run tests on.
+
+In its simplest form, `try-runtime` is a tool that enables:
+
+1. Connecting to a remote node and calling into some runtime API.
+2. Taking some or all of the data for that node at a specified block.
+3. Writing tests for that data.
+
+> **Note:** This tool is designed to be used for testing _before_ going to production.
+
+## Motivation
+
+The initial motivation for `try-runtime` came from the need to extend the capabilities of [`TestExternalities`][testextern-rustdocs] and [`BasicExternalities`][basicextern-rustdocs] 
+in such a way that it would be possible to use externalities to populate its memory with real chain's data. Since externalities provide the capability of storing an in-memory
+hash-map, extending these capabilities is achieved by simply using the key-value pairs from a node's RPCs.
+
+## How it works
+The `try-runtime` tool has its own implementation of externalities called [`remote_externalities`][remoteextern-rustdocs], with its hash-map containing:
```

Comment:
Not clear to me if its important that its a hash-map. I think if we want to go into details we can describe it more generically as a key value store where most data is scale encoded. We can link to both scale encoding and a section on keys.
Some of these are a little dated but they are good explainers when it comes to keys and storage:
- https://www.shawntabrizi.com/substrate/querying-substrate-storage-via-rpc/
- https://www.shawntabrizi.com/substrate/substrate-storage-deep-dive/
- https://www.shawntabrizi.com/substrate/transparent-keys-in-substrate/